odoo/tools/populate.py

```diff
@@ -59,12 +61,26 @@ def get_field_variation_date(model: Model, field: Field, factor: int, series_ali
     setting duplicated records too far back in the past.
     """
     total_days = min((MAX_DATETIME - MIN_DATETIME).days, factor)
-    return SQL("""
-        %(field)s - (%(factor)s - %(series_alias)s) * floor(%(total_days)s/%(factor)s) * interval '1 days'
-    """, field=SQL.identifier(field.name),
+    cast_type = SQL('date') if field.type == 'date' else SQL('timestamp')
+    query = SQL("""
+        (%(field)s - (%(factor)s - %(series_alias)s) * floor(%(total_days)s/%(factor)s) * interval '1 days')::%(cast_type)s
+    """, field=SQL('value::%s', SQL(cast_type)) if field.company_dependent else SQL.identifier(field.name),
```

Comment:
This is confusing.  First, `cast_type` is already an `SQL` object.  Second, where does "value" come from?